pbs_light.go

```diff
@@ -700,6 +693,58 @@ func loadDataCache(cfg *config.Configuration) (err error) {
 	return nil
 }
 
+func newPostgresDb(cfg *config.PostgresConfig) (*sql.DB, error) {
```

Comment:
I do hate to see new functions added to pbs_light.go. This should ideally go somewhere into /cache or /stored_requests, or into its own new section. I would lean towards /stored_requests, since /cache itself is legacy code.

The whole PR is a step towards making adding new databases easier, so it has that going for it. :)

Reply:
yeah. I'm not happy with those "conditionally `nil` sql.DB" variables floating around in `pbs_light.go`, but.... *shrugs*. Critical bugfix.

We can definitely clean it up more easily once the legacy stuff is gone.